framework/branch-key-store.md

```diff
@@ -390,9 +420,57 @@ The operation MUST call AWS KMS GenerateDataKeyWithoutPlaintext with a request c
 If the call to AWS KMS GenerateDataKeyWithoutPlaintext succeeds,
 the operation MUST use the `CiphertextBlob` as the wrapped Beacon Key.
 
+##### Wrapped Beacon Key Creation `v2`
+
+The operation MUST create a map of strings,
+the [branch key context for beacon keys](#beacon-key-branch-key-context).
+
+The operation MUST calculate the **SHA-384 Digest for the beacon key**
+by [serializing](../structures.md#serialization) the [branch key context for beacon keys](#beacon-key-branch-key-context);
+the serialization MUST be done according to the [encryption context serialization specification](../structures.md#serialization).
+
+The operation MUST call [AWS KMS API GenerateDataKey](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html).
+The call to AWS KMS GenerateDataKey MUST use the configured AWS KMS client to make the call.
+The operation MUST call AWS KMS GenerateDataKey with a request constructed as follows:
+
+- `KeyId` MUST be the `kms-arn`
+- `NumberOfBytes` MUST be 32.
+- `EncryptionContext` MUST be the `encryption-context`
+- `GrantTokens` MUST be this keystore's [grant tokens](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#grant_token).
+
+If the call to AWS KMS GenerateDataKey succeeds,
+the operation MUST use the `Plaintext` from GenerateDataKey result
+as the plain-text Beacon Key.
+
+The operation MUST concatenate the **SHA-384 Digest for the beacon key**
+with the plain-text Beacon Key,
+creating the **beacon plain-text tuple**.
+
+The operation MUST call [AWS KMS API Encrypt](https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html)
+with a request constructed as follows:
+
+- `Plaintext` **beacon plain-text tuple**
```

Comment:
Must be

Reply:
resolved at [757de04](https://github.com/awslabs/aws-encryption-sdk-specification/pull/297/commits/757de044819ab04e1e0835cea793987ba0a46990)